tiledb/sm/rest/rest_client.cc

```diff
@@ -102,6 +104,8 @@ Status RestClient::init(
 
   config_ = config;
   compute_tp_ = compute_tp;
+  memory_tracker_ = resources.create_memory_tracker();
+  memory_tracker_->set_type(MemoryTrackerType::REST_CLIENT);
```

Comment:
I was just about to comment that this looked super wrong since we should have an Array's MemoryTracker, but then it occurred to me we probably also use this in lots of other places not attached to an array. Lets add a comment to the effect of "Using  MemoryTrackerType::REST_CLIENT for now since this class is used in many places not directly tied to an array." or something similar. I wouldn't be surprised if we end up changing these types around in the future so a note here on why we picked a not entirely obvious type would be good.

Reply:
Done!